pallets/automation-time/src/lib.rs

```diff
@@ -460,6 +566,28 @@ pub mod pallet {
 			10_000
 		}
 
+		fn transfer(
+			sender: T::AccountId,
+			recipient: T::AccountId,
+			amount: BalanceOf<T>,
+			task_id: T::Hash,
+		) -> Weight {
+			// check for sufficient funds in the sender Balance
+			let sender_balance = <T as Config>::Currency::free_balance(&sender);
+			if sender_balance < amount {
+				Self::deposit_event(Event::InsufficientFunds { task_id });
+				return 10_000;
+			}
+			match <T as Config>::Currency::transfer(&sender, &recipient, amount, ExistenceRequirement::KeepAlive) {
```

Comment:
since you only care about the error you should use an if let for this

Reply:
ahh i changed this to be this instead. fixes my return issue
```suggestion
			match <T as Config>::Currency::transfer(&sender, &recipient, amount, ExistenceRequirement::KeepAlive) {
				Ok(_number)  => Self::deposit_event(Event::SuccesfullyTransferredFunds { task_id }),
        		        Err(_e) => Self::deposit_event(Event::TransferFailed { task_id }),
			};
```